src/RedisResourceManager.php

```diff
@@ -596,7 +595,7 @@ protected function normalizeLibOptionKey(&$key)
             throw new Exception\InvalidArgumentException("Unknown redis option '{$key}' ({$const})");
         }
         /** @var int $key */
-        $key = constant($const);
+        $key = (int) constant($const);
```

Comment:
IMO a cast is worse here: it obfuscates cases where non-numeric values are fetched

Reply:
All these constant values should be an integer I believe

Reply:
Then we should at least `assert()` instead :-)

Reply:
Yeah I'd like to do much much more to ensure these types, and I think we may need to just accept BC breaks and a new major to do so.

Reply:
e.g. `normalizePersistentId` method just casts the variable reference to a string and nothing else, when the param type is already set to be a string. So assuming that the types are correct, this method is redundant, but that may actually break usages which aren't currently checking for types.

Reply:
for now I'd really just recommend coding the assumption in form an assertion though

Reply:
Will do for this line